File: leetcode/AIcoding/5.Qwen-3/erlang_solutions/3267-count-almost-equal-pairs-ii.erl
-module(solution).
-export([count_almost_equal_pairs/1]).

count_almost_equal_pairs(Nums) ->
    Count = 0,
    N = length(Nums),
    lists:foldl(fun(I, Acc) ->
        lists:foldl(fun(J, Acc2) ->
            if
                I < J ->
                    A = lists:nth(I + 1, Nums),
                    B = lists:nth(J + 1, Nums),
                    Diff = abs(A - B),
                    if
                        Diff =< 2 -> Acc2 + 1;
                        true -> Acc2
                    end;
                true -> Acc2
            end
        end, Acc, lists:seq(I + 1, N - 1))
    end, Count, lists:seq(0, N - 1)).

File: leetcode/AIcoding/2.Gemini-2.0/c_solutions/3267-count-almost-equal-pairs-ii.c
#include <stdio.h>
#include <stdlib.h>
#include <string.h>

int countAlmostEqualPairs(int* nums, int numsSize, int diff) {
    int count = 0;
    for (int i = 0; i < numsSize; i++) {
        for (int j = i + 1; j < numsSize; j++) {
            if (abs(nums[i] - nums[j]) <= diff) {
                count++;
            }
        }
    }
    return count;
}